### PR TITLE
Fixes cigarette branding and taste

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
fix: Cigarette branding has been fixed.
add: Uplift Smooth brand cigarettes now taste minty as advertised.
del: You can no longer inject cigarette packets to inject all cigarettes
simultaneously.
/:cl:

- Makes subtypes of all the brands of cigarettes, and sets them to the
packets, removes the brand var and the special casing initializing that
had to be done for it.
- Adds "menthol", a do nothing drug chemical that tastes of mint to
Uplift Smooth (because they say they taste of mint, so we're fixing bugs
in the user's experience)
- Cigarettes now have a "list_reagents" var, for specifying their
initial reagents mix, like `reagent_container`

### CODE REVIEW

**Comment by Shapsy** [comment]
Why remove injecting the cig packet directly?

**Comment by coiax** [comment]
It was implemented in a really weird and hacky way, removing and replacing a cigarette multiple  times would add multiple does of the reagents in the packet itself.

**Comment by octareenroon91** [comment]
Menthol is also a cough suppressant, but that would definitely be a feature.

**Comment by PKPenguin321** [comment]
Improve don't remove

**Comment by PKPenguin321** [comment]
Also this feels like a feature freeze circumvention for the sake of bring a freeze circumvention

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 1]
```diff
@@ -104,24 +104,29 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
-	var/lit = 0
+	var/lit = FALSE
+	var/starts_lit = FALSE
 	var/icon_on = "cigon"  //Note - these are in masks.dmi not in cigarette.dmi
 	var/icon_off = "cigoff"
 	var/type_butt = /obj/item/weapon/cigbutt
 	var/lastHolder = null
 	var/smoketime = 300
 	var/chem_volume = 30
+	var/list_reagents = list("nicotine" = 15)
```
Comment (RemieRichards) [inline code/game/objects/items/weapons/cigs_lighters.dm]: b-but this isn't a list ;_;

typecast damnit!

**Comment by ExcessiveUseOfCobblestone** [comment]
> Also this feels like a feature freeze circumvention for the sake of bring a freeze circumvention

Him and kevinz....

**Comment by dannno** [comment]
"improve dont remove" is usually stupid and doesn't apply to things that are broken

**Pull request merged**